src/master-playlist-controller.js

```diff
@@ -1145,7 +1150,7 @@ export class MasterPlaylistController extends videojs.EventTarget {
     }
 
     // Blacklist this playlist
-    currentPlaylist.excludeUntil = Date.now() + (blacklistDuration * 1000);
+    currentPlaylist.excludeUntil = currentPlaylist.retryCount > this.maxPlaylistRetries ? Infinity : Date.now() + (blacklistDuration * 1000);
```

Comment:
perhaps we should just set `blacklistDuration` to `Infinity` above this rather than adding to this statement?